draft-ietf-tls-esni.md

```diff
@@ -216,64 +239,83 @@ ECHConfigs structure.
-and multiple sets of ECH parameters.
+The `ECHConfigs` structure contains one or more `ECHConfig` structures in
+decreasing order of preference. This allows a server to support multiple
+versions of ECH and multiple sets of ECH parameters.
 
-The ECHConfig structure contains the following fields:
+The `ECHConfig` structure contains the following fields:
 
 version
-: The version of the structure. For this specification, that value SHALL be
-0xff07. Clients MUST ignore any ECHConfig structure with a version they do not
-understand.
+: The version of ECH for which this configuration is used. Versioning is
+described below in {{versioning}}. Clients MUST ignore any `ECHConfig` structure
+with a version they do not support.
+
+length
+: The length, in bytes, of the next field.
 
 contents
 : An opaque byte string whose contents depend on the version of the structure.
-For this specification, the contents are an ECHConfigContents structure.
+For this specification, the contents are an `ECHConfigContents` structure.
+[[TODO: `contents` isn't a named field in the `ECHConfig` structure. I'm
+assuming `contents` is meant to be the label of the variant, but that's not what
+the spec says.]]
 
-The ECHConfigContents structure contains the following fields:
+The `ECHConfigContents` structure contains the following fields:
 
 public_name
-: The non-empty name of the entity trusted to update these encryption keys.
-This is used to repair misconfigurations, as described in
+: The non-empty name of client-facing server, i.e., the entity trusted to update
+these encryption keys. This is used to repair misconfigurations, as described in
 {{handle-server-response}}.
 
 public_key
-: The HPKE {{!I-D.irtf-cfrg-hpke}} public key which can be used by the client to
-encrypt the ClientHello.
+: The HPKE public key used by the client to encrypt ClientHelloInner.
 
 kem_id
-: The HPKE {{!I-D.irtf-cfrg-hpke}} KEM identifier corresponding to public_key.
-Clients MUST ignore any ECHConfig structure with a key using a KEM they do not
-support.
+: The HPKE KEM identifier corresponding to `public_key`. Clients MUST ignore any
+`ECHConfig` structure with a key using a KEM they do not support.
 
 cipher_suites
-: The list of HPKE {{!I-D.irtf-cfrg-hpke}} AEAD and KDF identifier pairs clients
-can use for encrypting the ClientHello.
+: The list of HPKE AEAD and KDF identifier pairs clients can use for encrypting
+ClientHelloInner.
 
 maximum_name_length
-: The largest name the server expects to support, if known. If this value is
-not known it can be set to zero, in which case clients SHOULD use the inner
-ClientHello padding scheme described below. That could happen if wildcard names
-are in use, or if names can be added or removed from the anonymity set during
-the lifetime of a particular resource record value.
+: The largest name the server expects to support, if known. If this value is not
+known it can be set to zero, in which case clients SHOULD use the padding scheme
```

Comment:
I think we should revert this change. The padding scheme specifically applies to the ClientHelloInner.

Reply:
Reverted.